decision/engine.go

```diff
@@ -240,12 +251,26 @@ func (e *Engine) MessageReceived(p peer.ID, m bsmsg.BitSwapMessage) error {
 			log.Debugf("wants %s - %d", entry.Cid, entry.Priority)
 			l.Wants(entry.Cid, entry.Priority)
 			if exists, err := e.bs.Has(entry.Cid); err == nil && exists {
-				e.peerRequestQueue.Push(entry.Entry, p)
 				newWorkExists = true
+				if blockSize, _ = e.bs.GetSize(entry.Cid); blockSize == -1 {
```

Comment:
I wouldnt throw away an error. If this actually errors, something seems pretty wrong. It's worthy of logging IMO

Reply:
There was a previous mention from @Stebalien that `GetSize` wouldn't fail, but maybe we could add a comment there explaining it, since yes, that silent error seems suspicious.

> This really shouldn't be necessary. GetSize should always work (even if it isn't cached).

Reply:
Sorry, that statement was a bit confusing. I meant that if the `Get` worked, the `GetSize` should have worked. That is, we can just call:

```go
size, err := e.bs.GetSize(entry.Cid)
if err != nil {
    /* we don't have it, *may* be an error, check if it's bs.ErrNotFound */
} else {
    // we have the block
}
```

We don't need to call `GetSize`, then `Get`, then `GetSize`.

Reply:
@whyrusleeping, @schomatis, @schomatis  I agree. 
I removed the 'GetSize, then Get, then GetSize sequence' and I'm now logging the error. 

This is done. I pushed out these changes and squashed into a single commit.